src/nvim/popupmenu.c

```diff
@@ -667,10 +667,23 @@ void pum_redraw(void)
   }
 
   int scroll_range = pum_size - pum_height;
+  if (fconfig.border) {
+    // For mouse menu, ensure border doesn't overflow screen
+    int mouse_menu = State != MODE_CMDLINE && pum_grid.zindex == kZIndexCmdlinePopupMenu;
+    if (mouse_menu && pum_border_width > 0) {
```

Comment:
This zindex condition for `mouse_menu` seems hacky. In Vim there is no need for `pum_redraw()` to special-case the mouse menu at all. Why is it needed in Nvim?

Reply:
Because it is handled uniformly after parsing pumborder, rather than in pum_position_at_mouse.

Reply:
Then it's not handled uniformly, as you have a special case for the mouse menu in `pum_redraw()`.